modules/standard/Buffers.chpl

```diff
@@ -140,18 +141,35 @@ module Buffers {
      :arg error: (optional) capture an error that was encountered instead of
                  halting on error
    */
-  proc bytes.bytes(len:int(64), out error:syserr) {
+  proc bytes.init(len:int(64), out error:syserr) {
+    this.home = here;
+    super.init();
     error = qbytes_create_calloc(this._bytes_internal, len);
     // The buffer is "retained" internally on creation, but only on success.
-    this.home = here;
   }
   pragma "no doc"
-  proc bytes.bytes(len:int(64)) {
-    var error:syserr = ENOERR;
-    error = qbytes_create_calloc(this._bytes_internal, len);
+  proc bytes.init(len:int(64)) {
+    this.home = here;
+    super.init();
+    var error:syserr = qbytes_create_calloc(this._bytes_internal, len);
     if error then try! ioerror(error, "in bytes constructor");
     // The buffer is retained internally on construction, but only on success.
-    this.home = here;
+  }
+
+
+  pragma "no doc"
+  proc bytes.init(x: bytes) {
+    if x.home == here {
+      this.home = here;
+      this._bytes_internal = x._bytes_internal;
+      super.init();
+      qbytes_release(x._bytes_internal);
```

Comment:
This looks different from the initCopy function - this performs a qbytes_release while the initCopy performs a qbytes_retain.  Was that intentional?

Reply:
Now I'm embarassed I didn't catch that...

Reply:
This relates back to my lack of confidence in this change.  For some reason, I was thinking that since the copyInit() returned `x`, the behavior needed to be different in a copy initializer that was creating a new value and tried this pattern based on how the Owned/Shared modules implement assignment.  When it passed testing I wasn't sure whether to feel like I got it right or whether testing just didn't exercise copies.  Based on Michael's feedback I'll change it to qbytes_retain(), assume that's right and re-test.